base/loading.jl

```diff
@@ -1738,6 +1753,104 @@ end
 
 const MAX_NUM_PRECOMPILE_FILES = Ref(10)
 
+module Linking
+
+const lld_path = Ref{String}()
+const lld_exe = Sys.iswindows() ? "lld.exe" : "lld"
+
+function __init__()
+    # Prefer our own bundled lld, but if we don't have one, pick it up off of the PATH
+    # If this is an in-tree build, `lld` will live in `tools`.  Otherwise, it'll be in `libexec`
+    for bundled_lld_path in (joinpath(Sys.BINDIR, Base.LIBEXECDIR, lld_exe),
+                             joinpath(Sys.BINDIR, "..", "tools", lld_exe),
+                             joinpath(Sys.BINDIR, lld_exe))
+        if isfile(bundled_lld_path)
+            lld_path[] = abspath(bundled_lld_path)
+            return
+        end
+    end
+    lld_path[] = something(Sys.which(lld_exe), lld_exe)
+    return
+end
+
+function lld()
+    return Cmd([lld_path[]])
+end
+
+function ld()
+    default_args = ``
+    @static if Sys.iswindows()
+        flavor = "link"
+    elseif Sys.isapple()
+        flavor = "darwin"
+        arch = Sys.ARCH == :aarch64 ? :arm64 : Sys.ARCH
+        default_args = `-arch $arch -lSystem -platform_version macos 10 11`
```

Comment:
`-L/usr/lib` is needed to find `-lSystem`
```suggestion
        default_args = `-arch $arch -lSystem -L/usr/lib -platform_version macos 10 11`
```